lambda_function.py

```diff
@@ -0,0 +1,126 @@
+        return e
+
+
+def verify_responses(website_to_test_expected_responses: dict[str, Website],
+                     request = http.request) -> dict[str, Website]:
+    for website in website_to_test_expected_responses.values():
+        expected_response = website.expected_response
+
+        match expected_response:
+            case ConnectTimeoutError():
+                http_status: int | Exception = get_response(request, website.url)
+                if type(http_status) == ConnectTimeoutError:
+                    website.received_expected_response = True
+                    website.actual_response = str(website.expected_response)  # just in case there is sensitive
+                    # information in error message
+                else:
+                    website.received_expected_response = False
+                    website.actual_response = str(http_status)
+            case int():
+                http_status: int | Exception = get_response(request, website.url)
+                website.received_expected_response = http_status == website.expected_response
+                website.actual_response = str(http_status)
+            case _:
+                raise ValueError(f"Unrecognised expected_response: {expected_response}")
+
+    return website_to_test_expected_responses
+
+
+def print_error_message(websites: dict[str, Website], print_error = print):
+    for website_name, website in websites.items():
+        error_message_in_json = {
+            "Status": "Failure",
+            "Website": website_name,
+            "Message": "This address is unexpectedly available",
+            "Expected Response": str(website.expected_response),
+            "Actual Response": website.actual_response
+        }
+
+        print_error(error_message_in_json)
+
+
+def print_success_message(preservica_website: Website, print_error = print):
+    success_message_in_json = {
+        "Status": "Success",
+        "Website": f"{preservica_website.name}",
+        "Message": f"{preservica_website.name} returned an expected response: {preservica_website.expected_response}"
+    }
+
+    print_error(success_message_in_json)
+
+
+def get_websites_with_errors(preservica_website: Website, rest_of_the_tested_websites: dict[str, Website]) -> dict[
+    str, Website]:
+    if preservica_website.received_expected_response:
+        any_other_website_received_unexpected_response = {
+            website for website in rest_of_the_tested_websites.values() if not website.received_expected_response
+        }
+
+        if any_other_website_received_unexpected_response:
+            print("Preservica website timed out as expected but other test websites did not receive expected response.")
+            return rest_of_the_tested_websites
+        return {}
+    else:
+        return {preservica_website.name: preservica_website}
+
+
+def lambda_handler(verify_responses_func = verify_responses, print_error_message_func=print_error_message,
```

Comment:
For the lambda to work properly, you need a function which takes (event, context) for the lambda to call.  
So here, if you rename this to `run_connection_tests` or something and then have:
```python
def lambda_handler(event, context):
    run_connection_tests()
```

Once that's in, you can get rid of the 
```python
if __name__ == "__main__":
    lambda_handler()
```
as you shouldn't need it.